typing/typecore.ml

```diff
@@ -133,6 +133,18 @@ type error =
 exception Error of Location.t * Env.t * error
 exception Error_forward of Location.error
 
+let trace_of_error = function
+    Label_mismatch (_,tr)
+  | Pattern_type_clash (tr,_)
+  | Or_pattern_type_clash (_,tr)
+  | Expr_type_clash (tr,_,_)
+  | Coercion_failure (_,_,tr,_)
+  | Less_general (_,tr)
+  | Letop_type_clash (_,tr)
+  | Andop_type_clash (_,tr)
+  | Bindings_type_clash tr -> Some tr
+  | _ -> None
```

Comment:
Let's make that non-fragile.

Reply:
You really want me to enumerate all the cases?
Note that this is not exhaustive anyway: there are type errors that do not include a trace, and errors that include a trace in other modules.

Reply:
I would have liked to.
But I don't strongly care.